File: Python/DSA/Sorting/selection_sort.py
def selection_sort(arr):
    size = len(arr)

    for i in range(size-1):
        min_idx = i
        for j in range(i+1,size):
            if arr[min_idx] > arr[j]:
                min_idx = j

         # Swap elements
        arr[i], arr[min_idx] = arr[min_idx], arr[i]

""" Tradional swapping
    temp = arr[i]
    arr[i] = arr[min_idx]
    arr[min_idx] = temp
"""


arr = [4,5,24,5,43,12,43,2,52,17]

selection_sort(arr)
print(arr)

File: Java/DSA/Sorting/SelectionSort.java
package Sorting;

import java.util.Arrays;

/**
 * Resources : [Programiz selection sort](https://www.programiz.com/dsa/selection-sort)
 */
public class SelectionSort {

    public static void main(String[] args) {
        int[] data = { 20, 12, 10, 15, 2 };
        selectionSort(data);
        System.out.println("Sorted Array in Ascending Order: ");
        System.out.println(Arrays.toString(data));
    }


    static void selectionSort(int[] arr){

        int n = arr.length;

        for(int i=0;i<n;i++){
            int min_idx = i;
            for(int j=i+1;j<n;j++){
                if(arr[j] < arr[min_idx]){
                    min_idx = j;
                }
            }

            int temp = arr[i];
            arr[i] = arr[min_idx];
            arr[min_idx] = temp;
        }

    }

}
